freqtrade/strategy/strategy_updater.py

```diff
@@ -0,0 +1,213 @@
+                return node
+
+            def visit_FunctionDef(self, node):
+                # if the function name is in the mapping, update it
+                if node.name in _function_mapping:
+                    node.name = _function_mapping[node.name]
+                return self.generic_visit(node)
+
+            def visit_Attribute(self, node):
+                # if the attribute name is 'nr_of_successful_buys',
+                # update it to 'nr_of_successful_entries'
+                if isinstance(node.value, ast.Name) and \
+                        node.value.id == 'trades' and \
+                        node.attr == 'nr_of_successful_buys':
+                    node.attr = 'nr_of_successful_entries'
+                return self.generic_visit(node)
+
+            def visit_ClassDef(self, node):
+                # check if the class is derived from IStrategy
+                if any(isinstance(base, ast.Name) and
+                       base.id == 'IStrategy' for base in node.bases):
+                    # check if the INTERFACE_VERSION variable exists
+                    has_interface_version = any(
+                        isinstance(child, ast.Assign) and
+                        isinstance(child.targets[0], ast.Name) and
+                        child.targets[0].id == 'INTERFACE_VERSION'
+                        for child in node.body
+                    )
+
+                    # if the INTERFACE_VERSION variable does not exist, add it as the first child
+                    if not has_interface_version:
+                        node.body.insert(0, ast.parse('INTERFACE_VERSION = 3').body[0])
+                    # otherwise, update its value to 3
+                    else:
+                        for child in node.body:
+                            if isinstance(child, ast.Assign) and \
+                                    isinstance(child.targets[0], ast.Name) and \
+                                    child.targets[0].id == 'INTERFACE_VERSION':
+                                child.value = ast.parse('3').body[0].value
+                return self.generic_visit(node)
+
+            def visit_Subscript(self, node):
+                if isinstance(node.slice, ast.Constant):
+                    if node.slice.value in strategy_updater.rename_dict:
+                        # Replace the slice attributes with the values from rename_dict
+                        node.slice.value = strategy_updater.rename_dict[node.slice.value]
+                if hasattr(node.slice, "elts"):
+                    for elt in node.slice.elts:
+                        if isinstance(elt, ast.Constant) and \
+                                elt.value in strategy_updater.rename_dict:
+                            elt.value = strategy_updater.rename_dict[elt.value]
+                return node
+
+            def visit_Constant(self, node):
+                # do not update the names in import statements
+                if node.value in \
```

Comment:
we should try to not wrap lines with `\` (if possible, obviously).
i've seen that break in mixed windows/linux environments - so it suddenly simply no longer works - therefore i became somewhat allergic against this :laughing: 

in if statements, it's better to wrap it all in parentheses `()`  (obviously not just in this line)

Reply:
those \ ones were primarily just cause either pycharm or github argued about too many characters in a row ...
yeah will be fixing that as much as possible
to me, who cares if a row is 105 characters instead of max 100